src/postgres/src/backend/access/index/genam.c

```diff
@@ -439,7 +439,16 @@ systable_getnext(SysScanDesc sysscan)
 
 	if (IsYugaByteEnabled())
 	{
-		return ybc_systable_getnext(sysscan);
+		const int64 old_query_id = MyProc->queryid;
+		if (old_query_id == 0) {
+			const int64 kCatalogQueryId = -3;
+			YBCSetQueryId(kCatalogQueryId);
+		}
+		HeapTuple ret = ybc_systable_getnext(sysscan);
+		if (old_query_id == 0) {
+			YBCSetQueryId(0);
+		}
+		return ret;
```

Comment:
There can be catalog requests after the query id is set. See this trace: http://10.150.1.75:16686/trace/09697fa78beea461028531060aaa06c1?uiFind=dd35d7791d4900f8

Should they be marked as -3 as well?

Also, query_id can be incorrectly set to 0 after the query_id has already been  calculated.

Reply:
`Also, query_id can be incorrectly set to 0 after the query_id has already been calculated.`
Don't think so. Unless the argument is that query_id is being set inside this function.

We are essentially resetting the QueryId to whatever it was earlier at line 442.

Reply:
There are catalog requests which are part of a statement but for which the query_id has not been calculated yet. Similarly there are catalog requests which are made after query_id is calculated. Isn't it better for the query_id to be consistent across them?

Reply:
As discussed, I  think it is better to use the query-id for catalog requests -- when that is available, than use a random/fixed "-3". 

For the ones where we are making catalog-requests from places where we have no query-id, it is good to tag them with a different query-id so we can document it as a "known" query/rpc. Will help us catch any new/other catalog requests that may be made -- which we  might not have caught.

Reply:
Agreed